reframe/core/logging.py

```diff
@@ -425,10 +426,11 @@ def _update_check_extras(self):
         if self.check.job:
             self.extra['check_jobid'] = self.check.job.jobid
 
-    def log_performance(self, level, tag, value, ref,
+    def log_performance(self, level, num_tasks, tag, value, ref,
                         low_thres, upper_thres, unit=None, *, msg=None):
 
         # Update the performance-relevant extras and log the message
+        self.extra['check_num_tasks'] = num_tasks
```

Comment:
This should not be here. It's a general check attribute and it should be valid for all the logging. It should be set in the `LoggerAdapted._update_check_extras()`. In the performance log format you can use any check's extra attribute. So remove also the `num_tasks` from the signature of this function.